explore_simulation_approach/1_simulate_and_identify_generic_genes_using_noise_approach.py

```diff
@@ -0,0 +1,349 @@
+#                        project_id,
+#                        simulated_data_filename,
+#                        "simulated",
+#                        local_dir,
+#                        i)
+# }
+# -
+
+# ## Rank genes
+
+analysis_type = "DE"
+template_DE_stats, simulated_DE_summary_stats = ranking.process_and_rank_genes_pathways(
+    template_DE_stats_filename,
+    local_dir,
+    num_runs,
+    project_id,
+    analysis_type,
+    col_to_rank_genes,
+    logFC_name,
+    pvalue_name,
+)
+
+# ## Gene summary table
+#
+# Note: Using DESeq, genes with NaN in `Adj P-value (Real)` column are those genes flagged because of the `cooksCutoff` parameter. The cook's distance as a diagnostic to tell if a single sample has a count which has a disproportionate impact on the log fold change and p-values. These genes are flagged with an NA in the pvalue and padj columns of the result table. For more information you can read [DESeq FAQs](https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#pvaluesNA)
+
+# +
+summary_gene_ranks = ranking.generate_summary_table(
+    template_DE_stats_filename,
+    template_DE_stats,
+    simulated_DE_summary_stats,
+    col_to_rank_genes,
+    local_dir,
+    "gene",
+    params,
+)
+
+summary_gene_ranks.head()
+# -
+
+summary_gene_ranks.isna().any()
+
+# Create `gene_summary_filename`
+summary_gene_ranks.to_csv(gene_summary_filename, sep="\t")
+
+# ## Compare gene ranking
+# Studies have found that some genes are more likely to be differentially expressed even across a wide range of experimental designs. These *generic genes* are not necessarily specific to the biological process being studied but instead represent a more systematic change.
+#
+# We want to compare the ability to detect these generic genes using our method vs those found by [Crow et. al. publication](https://www.pnas.org/content/pnas/116/13/6491.full.pdf). Their genes are ranked 0 = not commonly DE; 1 = commonly DE. Genes by the number differentially expressed gene sets they appear in and then ranking genes by this score.
+
+# +
+# Get generic genes identified by Crow et. al.
+DE_prior_filename = params["reference_gene_filename"]
+ref_gene_col = params["reference_gene_name_col"]
+ref_rank_col = params["reference_rank_col"]
+
+figure_filename = f"gene_ranking_{col_to_rank_genes}.svg"
+
+corr, shared_ranking = ranking.compare_gene_ranking(
+    summary_gene_ranks, DE_prior_filename, ref_gene_col, ref_rank_col, figure_filename
+)
+
+# +
+# Hypergeometric test:
+# Given N number of genes with K common genes in Crow et al.
+# SOPHIE identifies n genes as being common
+# What is the probability that k of the genes identified by SOPHIE
+# are also common in Crow et al.? What is the probability of drawing
+# k or more concordant genes?
+
+num_Crow_genes = shared_ranking.shape[0]
+num_generic_Crow_genes = shared_ranking.query(f"{ref_rank_col}>=80.0").shape[0]
+num_generic_SOPHIE_genes = shared_ranking[
```

Comment:
Are "SOPHIE genes" here genes identified as generic after adding Gaussian noise? I would think SOPHIE would involve your VAE/latent space shift method, but maybe I'm wrong.

Reply:
This is definitely a "copy and paste" error:) I will fix this